test/test_helpers.py

```diff
@@ -319,5 +319,35 @@ def test_get_cartesian_face_edge_nodes_filled_value(self):
             'North', face_edges_connectivity_cartesian)
 
         # Assert that the result is True
+        self.assertTrue(result)
+
+    def test_get_lonlat_face_edge_nodes_pipeline(self):
+        # Create the vertices for the grid, based around the North Pole
+
+        vertices = [[0.5, 0.5, 0.5], [-0.5, 0.5, 0.5], [-0.5, -0.5, 0.5],
+                    [0.5, -0.5, 0.5]]
+
+        #Normalize the vertices
+        vertices = [x / np.linalg.norm(x) for x in vertices]
+
+        # Construct the grid from the vertices
+        grid = ux.Grid.from_face_vertices(vertices, latlon=False)
+        face_edges_connectivity_lonlat = _get_lonlat_rad_face_edge_nodes(
+            grid.face_node_connectivity.values[0],
+            grid.face_edge_connectivity.values[0],
+            grid.edge_node_connectivity.values, grid.node_lon.values,
+            grid.node_lat.values)
+
+        # Convert all the values into cartesian coordinates
+        face_edges_connectivity_cartesian = []
+        for i in range(len(face_edges_connectivity_lonlat)):
+            edge = face_edges_connectivity_lonlat[i]
+            edge_cart =   [node_lonlat_rad_to_xyz(node) for node in edge]
+            face_edges_connectivity_cartesian.append(edge_cart)
```

Comment:
I am open to any suggestions for the code style. But again, I don't have much time doing the code refractory. Maybe you and @philipc2 can help me turn this codes into the desired format? I think it can be done with the optimization steps

Reply:
Sounds good. I am copying this comment over to #730 then